src/components/Search.jsx

```diff
@@ -18,9 +18,20 @@ function Search({ intl }) {
   const { page } = useContext(DiscussionContext);
   const postSearch = useSelector(({ threads }) => threads.filters.search);
   const topicSearch = useSelector(({ topics }) => topics.filter);
+  const learnerSearch = useSelector(({ learners }) => learners.usernameSearch);
   const isPostSearch = ['posts', 'my-posts'].includes(page);
+  const isTopicSearch = 'topics'.includes(page);
   let searchValue = '';
 
+  let currentValue = '';
+  if (isPostSearch) {
+    currentValue = postSearch;
+  } else if (isTopicSearch) {
+    currentValue = topicSearch;
+  } else {
+    currentValue = learnerSearch;
+  }
+
```

Comment:
convert into memoize function. Otherwise, this calculation will happen on every render.

const currentValue = useMemo(()=> {
let value = '';
  if (isPostSearch) {
    value = postSearch;
  } else if (isTopicSearch) {
    value = topicSearch;
  } else {
    value = learnerSearch;
  }
return value;
}, [isPostSearch, isTopicSearch])